# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../squashfs-tools/hotfix.patch.musl
# Copyright (C) 2019 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- squashfs4.3/squashfs-tools/pseudo.c.vanilla	2019-02-09 09:57:25.466468224 +0000
+++ squashfs4.3/squashfs-tools/pseudo.c	2019-02-09 09:57:34.982468116 +0000
@@ -32,6 +32,7 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <sys/stat.h>
 #include <ctype.h>
 
 #include "pseudo.h"
--- squashfs4.3/squashfs-tools/mksquashfs.c.vanilla	2019-02-09 09:57:42.634468029 +0000
+++ squashfs4.3/squashfs-tools/mksquashfs.c	2019-02-09 09:57:56.082467875 +0000
@@ -4403,7 +4403,7 @@
 			regexec(path->name[i].preg, name, (size_t) 0,
 					NULL, 0) == 0 :
 			fnmatch(path->name[i].name, name,
-				FNM_PATHNAME|FNM_PERIOD|FNM_EXTMATCH) == 0;
+				FNM_PATHNAME|FNM_PERIOD/*|FNM_EXTMATCH*/) == 0;
 
 		if(match) {
 			 if(path->name[i].paths == NULL || new == NULL)
--- squashfs4.3/squashfs-tools/unsquashfs.c.vanilla	2019-02-09 09:58:06.806467753 +0000
+++ squashfs4.3/squashfs-tools/unsquashfs.c	2019-02-09 09:58:20.270467599 +0000
@@ -1429,7 +1429,7 @@
 			int match = use_regex ?
 				regexec(path->name[i].preg, name, (size_t) 0,
 				NULL, 0) == 0 : fnmatch(path->name[i].name,
-				name, FNM_PATHNAME|FNM_PERIOD|FNM_EXTMATCH) ==
+				name, FNM_PATHNAME|FNM_PERIOD/*|FNM_EXTMATCH*/) ==
 				0;
 			if(match && path->name[i].paths == NULL)
 				/*
--- squashfs4.3/squashfs-tools/action.c.vanilla	2019-02-09 09:58:31.074467476 +0000
+++ squashfs4.3/squashfs-tools/action.c	2019-02-09 09:59:06.242467075 +0000
@@ -1908,11 +1908,11 @@
 
 TEST_FN(name, ACTION_ALL_LNK, \
 	return fnmatch(atom->argv[0], action_data->name,
-				FNM_PATHNAME|FNM_PERIOD|FNM_EXTMATCH) == 0;)
+				FNM_PATHNAME|FNM_PERIOD/*|FNM_EXTMATCH*/) == 0;)
 
 TEST_FN(pathname, ACTION_ALL_LNK, \
 	return fnmatch(atom->argv[0], action_data->subpath,
-				FNM_PATHNAME|FNM_PERIOD|FNM_EXTMATCH) == 0;)
+				FNM_PATHNAME|FNM_PERIOD/*|FNM_EXTMATCH*/) == 0;)
 
 
 static int count_components(char *path)
@@ -1955,7 +1955,7 @@
 {
 	return fnmatch(atom->argv[0], get_start(strdupa(action_data->subpath),
 		count_components(atom->argv[0])),
-		FNM_PATHNAME|FNM_PERIOD|FNM_EXTMATCH) == 0;
+		FNM_PATHNAME|FNM_PERIOD/*|FNM_EXTMATCH*/) == 0;
 }
 
 TEST_VAR_FN(filesize, ACTION_REG, action_data->buf->st_size)
